### PR TITLE
Message creation changes in check_result

### LINKED ISSUE #44: 3 parts to a message
praise/encouragement + question specific + correct/incorrect message

e.g.,

```
             question specific (e.g., from pass_if)
             v
"You did it! That's the correct answer. Next, we'll see how we can apply this to dataframes."
^                                       ^
praise/encouragement                    correct message
```

There's currently only 2.  The "praise/encouragement" is overwritten by "correct".

### LINKED ISSUE #51: all params supplied to any glue statement should start with a `.`
`.message` etc

### PR DESCRIPTION
**Pull request by chendaniely**
PR addressing how messages are created and fixes #44 
Fixes #51 

Internal meeting notes from 2019-07-24:

For the correct/incorrect argument:
- options(grader_glue_correct = “{ praise() } { .message } { .correct }”) 
- options(grader_glue_incorrect = “{ .message } { .incorrect }{ encourage() }”) 
- And then check_result gets two new arguments glue_correct = getOption(...), glue_incorrect = getOption(...).
- The default values for correct and incorrect is NULL.  Internally, NULL string values should be upgraded to “” (the empty string) to work properly with `glue`

### CODE REVIEW

File: R/check_result.R [hunk 1]
```diff
@@ -56,9 +66,11 @@ check_result <- function(
     list(
       matched = found_match,
       correct = final_result$correct,
-      message = final_result$message
+      .message = final_result$message,
+      .correct = correct,
```
Comment (schloerke) [inline R/check_result.R]: remove

File: R/check_result.R [hunk 2]
```diff
@@ -56,9 +66,11 @@ check_result <- function(
...
+      .incorrect = incorrect
```
Comment (schloerke) [inline R/check_result.R]: remove

File: R/check_result.R [hunk 3]
```diff
@@ -56,9 +66,11 @@ check_result <- function(
     list(
       matched = found_match,
```
Comment (schloerke) [inline R/check_result.R]: change to `.matched`

File: R/check_result.R [hunk 4]
```diff
@@ -56,9 +66,11 @@ check_result <- function(
...
       correct = final_result$correct,
```
Comment (schloerke) [inline R/check_result.R]: change to `.correct`

**Comment by chendaniely** [comment]
@schloerke I'm merging this into `dev` so I can rebase #47 against it.

**Pull request merged**